File: general-workshop/code/scala/02-data-eng/06-BatchJob/GlobalVarsAndMethods.scala
// Databricks notebook source
//Database credentials & details - for use with Spark scala for writing

// Secrets
val jdbcUsername = dbutils.secrets.get(scope = "gws-sql-db", key = "username")
val jdbcPassword = dbutils.secrets.get(scope = "gws-sql-db", key = "password")

// JDBC driver class & connection properties
val driverClass = "com.microsoft.sqlserver.jdbc.SQLServerDriver"
val jdbcHostname = "gws-server.database.windows.net"
val jdbcPort = 1433
val jdbcDatabase = "gws_sql_db"

// JDBC URI
val jdbcUrl = s"jdbc:sqlserver://${jdbcHostname}:${jdbcPort};database=${jdbcDatabase}"

// Properties() object to hold the parameters
import java.util.Properties
val connectionProperties = new Properties()

connectionProperties.put("user", s"${jdbcUsername}")
connectionProperties.put("password", s"${jdbcPassword}")
connectionProperties.setProperty("Driver", driverClass)

// COMMAND ----------

def generateBatchID(): Int = 
{
  var batchId: Int = 0
  var pushdown_query = "(select count(*) as record_count from BATCH_JOB_HISTORY) table_record_count"
  val df = spark.read.jdbc(url=jdbcUrl, table=pushdown_query, properties=connectionProperties)
  val recordCount = df.first().getInt(0)
  println("Record count=" + recordCount)
  
  if(recordCount == 0)
    batchId=1
  else 
  {
    pushdown_query = "(select max(batch_id) as current_batch_id from BATCH_JOB_HISTORY) current_batch_id"
    val df = spark.read.jdbc(url=jdbcUrl, table=pushdown_query, properties=connectionProperties)
    batchId = df.first().getInt(0) + 1
  }
  batchId
}

File: data-engineering/nyc-taxi-workshop/code/notebooks-py/06-BatchJob/GlobalVarsAndMethods.py
# Databricks notebook source
# MAGIC %sql
# MAGIC 
# MAGIC CREATE DATABASE IF NOT EXISTS <userid>_taxi_reports_db;
# MAGIC USE <userid>_taxi_reports_db;
# MAGIC 
# MAGIC DROP TABLE IF EXISTS BATCH_JOB_HISTORY;
# MAGIC CREATE TABLE BATCH_JOB_HISTORY
# MAGIC USING org.apache.spark.sql.jdbc
# MAGIC OPTIONS (
# MAGIC   url 'jdbc:sqlserver://<yourDatabaseServer>.database.windows.net:1433;database=<yourDatabase>;',
# MAGIC   dbtable 'BATCH_JOB_HISTORY',
# MAGIC   user '<yourUserID>',
# MAGIC   password "yourPassword"
# MAGIC );

# COMMAND ----------

#1) JDBC driver class & connection properties
connectionProperties = {
    "driver": "com.microsoft.sqlserver.jdbc.SQLServerDriver"
}

#JDBC URL
jdbcUrl = ("jdbc:sqlserver://<yourDatabaseServer>.database.windows.net:1433;database=<yourDatabase>;user=<yourUserID>;password=yourPassword")

# COMMAND ----------

def generateBatchID():
  batchId = 0
  recordCount = sql("select count(*) from <userid>_taxi_reports_db.BATCH_JOB_HISTORY").first()[0]
  print("Record count={}".format(recordCount))

  if(recordCount == 0):
    batchId=1
  else:
    batchId= sql("select max(batch_id) from <userid>_taxi_reports_db.BATCH_JOB_HISTORY").first()[0] + 1
 
  return batchId
